core/src/main/java/org/frankframework/parameters/NumberParameter.java

```diff
@@ -110,31 +112,29 @@ public Object getValue(ParameterValueList alreadyResolvedParameters, Message mes
 	}
 
 	@Override
-	protected Number getValueAsType(Object request, boolean namespaceAware) throws ParameterException, IOException {
+	protected Number getValueAsType(@Nonnull Message request, boolean namespaceAware) throws ParameterException, IOException {
 		if(getType() == ParameterType.NUMBER) {
-			if (request instanceof Number number) {
+			if (request.asObject() instanceof Number number) {
```

Comment:
Deze is deprecated (met een hele goede reden).
Hier moet je `message.isRequestOfType(Number.class)` gebruiken.

Reply:
Dat wou ik ook doen maar ik heb hem ook nodig. Dus dat kon uiteindelijk niet zonder `asObject()`.

Om het argument naar `valueAsType()` een `Message` te kunnen maken, moest ik de value wrappen en weer kunnen unwrappen. En de value unwrappen moet sowieso af en toe, eigenlijk -- de waarde kan een scalar database query result zijn die als `Message` in de session gestopt is bijvoorbeeld (dat is het scenario waarop de vorige change fout ging).

We kunnen van de Date of Number een String van maken en die parsen, of we kunnen die direct unwrappen.

Een betere oplossing zie ik voorlopig niet! Als jij suggesties hebt, houdt ik me aanbevolen.